units/pod_termination_test.go

```diff
@@ -156,7 +158,7 @@ func TestPodTerminationJob(t *testing.T) {
 			dbTask, err := task.FindOneId(tsk.Id)
 			require.NoError(t, err)
 			require.NotZero(t, dbTask)
-			assert.Equal(t, evergreen.TaskUndispatched, dbTask.Status, "stranded task should have been restarted")
+			assert.True(t, dbTask.ShouldAllocateContainer(), "stranded task should have been restarted to re-attempt allocation")
```

Comment:
can we check `ContainerAllocated` field on the task as well?

Reply:
`ShouldAllocateContainer` checks `ContainerAllocated` because it'll return false if it's already been allocated a container.